src/power_policy/_power_policy.cpp

```diff
@@ -20,17 +22,30 @@
 #define ARR_ARGS_2(arr_name) ARR_ARGS_1(arr_name), arr_name[1]
 #define ARR_ARGS_3(arr_name) ARR_ARGS_2(arr_name), arr_name[2]
 #define ARR_ARGS_4(arr_name) ARR_ARGS_3(arr_name), arr_name[3]
-#define PP(name, class_, arg_count)                                                                \
-    do {                                                                                           \
-        if (policy_name == name) {                                                                 \
-            if (arg_count != policy_args.size()) { /* NOLINT(readability-container-size-empty) */  \
-                throw runtime_error(                                                               \
-                  "Incorrect number of parameters for the '" name "' power policy; got " +         \
-                  std::to_string(policy_args.size()) + ", expected " #arg_count);                  \
-            }                                                                                      \
-            return std::make_unique<class_>(ARR_ARGS_##arg_count(policy_args));                    \
-        }                                                                                          \
-    } while (0)
+#define PP(name, class_, arg_count)                                                                     \
+    { /* this is std::pair */                                                                           \
+        name, [](const std::vector<std::string> &policy_args) {                                         \
+            if (arg_count != policy_args.size()) { /* NOLINT(readability-container-size-empty) */       \
+                throw runtime_error(                                                                    \
+                    "Incorrect number of parameters for the '" name "' power policy; got " +            \
+                    std::to_string(policy_args.size()) + ", expected " #arg_count);                     \
+            }                                                                                           \
+            return std::make_unique<class_>(ARR_ARGS_##arg_count(policy_args));                         \
+        }                                                                                               \
+    }
+
+static const std::map<std::string, std::function<std::unique_ptr<PowerPolicy>(std::vector<std::string>)>> power_policies {
```

Comment:
Missing `const ... &` (I think the vector gets copied, and then reference to the copy passed to the lambda)

Reply:
Probably yes, thanks!